sbbbattlesim/player.py

```diff
@@ -299,3 +303,34 @@ def cast_spell(self, spell_id, trigger_onspell=True):
             stack = self('OnSpellCast', caster=self, spell=spell, target=target)
 
         spell(self).cast(target=target)
+
+    def to_state(self):
+        # Need to case the tribes from enum to their strings
+        treasures = [key for key, value in self.treasures.items() for treasure in value ]
+
+        characters = [
+            {
+                'position': slot,
+                'id': character.id,
+                'attack': character.attack,
+                'health': character.health,
+                'golden': character.golden,
+                'cost': character.cost,
+                'tribes': [tribe.name.lower() for tribe in character.tribes],
+            }
+            for slot, character in self.__characters.items()
+            if character is not None
+        ]
+        hero = self.hero.id
+        spells = list(self.spells)
+        return {
+            'characters': characters,
+            'treasures': treasures,
+            'hero': hero,
+            'spells': spells,
+            'hand': self.hand,
```

Comment:
No changes are needed but I'm leaving this note to remind me that if/when hand is tracked it will be a list of characters and this will need to be updated.

Reply:
should I add a comment in the code about this?

Reply:
It's fine for now just wanted to leave a note somewhere